### PR TITLE
add flip function and tests

### LINKED ISSUE #36: Add a `flip()` function
I've been reading issue #16 (closed now), and just wanted to bump the idea of having a `flip()` function.

The original proposition by @jfsiii:

> ### `flip` (or `reverseArguments`, etc)
> 
> ```js
> function flip(fn) {
>   return function() {
>     var args = [].slice.call(arguments).reverse();
>     return fn.apply(this, args);
>   };
> }
> ```
> 
> It might seem silly to have a function which just reverses argument order, but it's a great help when you want to invoke a existing function which does what you want/need but has arguments in the opposite order.

## Why it's great

In addition to the arguments posted already, I see one great use case – currying right-hand-side arguments.

In the JS world this is common: `doGreatStuff(target, options)`. A function with optional arguments or an `options` object listed on the very right.

I often want to `.bind()` the optional arguments. Wouldn't it be great to do `flip(doGreatStuff).bind(null, {candyFloss: true})`?

### PR DESCRIPTION
**Pull request by stoeffel**
closes #36

### CODE REVIEW

File: flip.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * Returns a function with flipped arguments
+ * @function module:101/flip
+ * @param {function} f - function to be flipped
+ * @return {function}
+ */
+module.exports = flip;
+
+function flip(f) {
+  return function() {
+    var args = [].slice.call(arguments);
```
Comment (bkendall) [inline flip.js]: very much a nit, but to be consistent with the rest of the code base, could you swap out `[].slice` for `Array.prototype.slice`?

Comment (stoeffel) [inline flip.js]: Sure, I will change it.

Comment (tjmehta) [inline flip.js]: Yup, for consistency. Side note here is that using the prototype does not create a new array instance just for the purposes of using slice.

**Comment by tjmehta** [comment]
Hey I realized this too quick, but I pulled in a PR for upgrading to Lab 5. This may affect your tests as Lab's api has changed slightly and we are not using chaijs for assertions anymore. We are now using [hapijs/code](https://github.com/hapijs/code) it's syntax is *very* similar to chai's so it will probably only need some minor changes.

Just check out the other test files after syncing with master, and you should be fine.

**Comment by stoeffel** [comment]
@tjmehta I updated my tests

**Comment by bkendall** [comment]
LGTM :)

**Comment by tjmehta** [comment]
:+1:

**Pull request merged**